### DISCUSSION TITLE
Helmv3 Lock file Auth

### DISCUSSION THREAD

**Discussion by robgordon89** [renovatebot/renovate#17567] [Request Help]
I seem to be having an issue with Helm lock updates

```
File name: helm/Chart.lock

Error: looks like "https://raw.githubusercontent.com/REDACTED/charts/main/index" is not a valid chart repository or cannot be reached: failed to fetch https://raw.githubusercontent.com/REDACTED/charts/main/index/index.yaml : 404 Not Found
```

The PRs are created but not complete the lock file is not updated. I'm confused as the auth defiantly works as the PR's are created with the correct versions in Chart.yaml but fails to update the Chart.lock file when running `helm dependency update` 

Below are my hostRules 

```json
 {
    "matchHost": "https://raw.githubusercontent.com/REDACTED/charts",
    "hostType":"helm",
    "username": "REDACTED",
    "encrypted" :
    {
        "password": "..."
    }
 }
```

**Comment by rarkins** [comment]
It's a private repo?

**Reply by robgordon89** [reply]
yes correct private repo

**Reply by rarkins** [reply]
Setting aside Renovate for a moment, how would you normally configure credentials into `helm` so that `helm dependency updates` succeeds for private repositories?

The helmv3 artifacts updating logic includes some logic for hostRules but only seems to be for images: https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/helmv3/artifacts.ts

**Reply by robgordon89** [reply]
Adding the repo:

```
helm repo add --username {username} --password {personal_access_token} REDACTED https://raw.githubusercontent.com/REDACTED/charts/main/index
```

This creates an entry in `/Users/{USER}/.config/helm/repositories.yaml` with the relevant username/password 

Looks like this is run here https://github.com/renovatebot/renovate/blob/f1cbbd1a807d98a52db1529c0a437b9aa6d87a27/lib/modules/manager/helmv3/artifacts.ts#L78

**Reply by rarkins** [reply]
Sounds like we will need to add this capability. Can you create a feature request with minimal reproduction? The actual URL can be a fake or non-existent org, and the credentials should be dummy ones, but we want it to otherwise be correct enough that a developer can swap in a real private repo and credentials and reproduce

**Reply by robgordon89** [reply]
i will try and do this today 👍 thanks for looking

**Reply by robgordon89** [reply]
added a feature request here https://github.com/renovatebot/renovate/issues/17573

Thanks
